aeron-agent/src/main/java/io/aeron/agent/ArchiveEventEncoder.java

```diff
@@ -29,6 +29,198 @@ private ArchiveEventEncoder()
     {
     }
 
+    static <E extends Enum<E>> int encodeReplaySessionStateChange(
+        final UnsafeBuffer encodingBuffer,
+        final int offset,
+        final int captureLength,
+        final int length,
+        final E from,
+        final E to,
+        final long id,
+        final long recordingId,
+        final long position,
+        final String reason)
+    {
+        int encodedLength = encodeLogHeader(encodingBuffer, offset, captureLength, length);
+
+        encodingBuffer.putLong(offset + encodedLength, id, LITTLE_ENDIAN);
+        encodedLength += SIZE_OF_LONG;
+        encodingBuffer.putLong(offset + encodedLength, recordingId, LITTLE_ENDIAN);
+        encodedLength += SIZE_OF_LONG;
+        encodingBuffer.putLong(offset + encodedLength, position, LITTLE_ENDIAN);
+        encodedLength += SIZE_OF_LONG;
+
+        encodedLength += encodingBuffer.putStringAscii(offset + encodedLength, reason);
```

Comment:
Because we have two variable length strings to be encoded, we should consider the total length of the strings to determine if the data needs to be truncated.  It would probably be safer to put the state change before the reason text as the state description is likely to be shorter.